File: solutions/cpp/matching-brackets/1/matching_brackets.cpp
#include "matching_brackets.h"
#include <string_view>
#include <map>
#include <set>
#include <stack>
namespace matching_brackets {

// TODO: add your solution here
    const std::map<char, char> MATCHING_BRACKET{
        {']', '['},
        {'}', '{'},
        {')', '('},
    };
    bool check(const std::string_view& data) {
        std::stack<char> brackets{};
        for (const auto ch : data) {
            switch (ch) {
            case '[': case '{': case '(':
                brackets.push(ch);
                break;

            case ']': case '}': case ')':
                if (!brackets.empty() && brackets.top() == MATCHING_BRACKET.at(ch))
                    brackets.pop();
                else
                    return false;
            }
        }
        return brackets.empty();
    }
}  // namespace matching_brackets

File: solutions/python/matching-brackets/1/matching_brackets.py
def is_paired(input_string: str) -> bool:
    stack: list[str] = []
    MATCHING_BRACKETS: dict[str, str] = {
        "}": "{",
        "]": "[",
        ")": "("
    }
    for char in input_string:
        if char in MATCHING_BRACKETS.values():
            stack.append(char)
        elif char in MATCHING_BRACKETS.keys():
            if stack == [] or MATCHING_BRACKETS[char] != stack.pop():
                return False

    return stack == []